eng/helix/helix.proj

```diff
@@ -90,32 +92,39 @@
 
     <PropertyGroup>
       <SharedFxVersion>@(_ResolvedProductVersionInfo->'%(PackageVersion)')</SharedFxVersion>
+      <_AppRefSubPath>packs\Microsoft.AspNetCore.App.Ref\$(SharedFxVersion)</_AppRefSubPath>
+      <_AppRuntimeSubPath>shared\Microsoft.AspNetCore.App\$(SharedFxVersion)</_AppRuntimeSubPath>
+      <_AppRefPath>$(RepoRoot).dotnet\$(_AppRefSubPath)</_AppRefPath>
+      <_AppRuntimePath>$(RepoRoot).dotnet\$(_AppRuntimeSubPath)</_AppRuntimePath>
+
+      <!--
+        Don't bother checking for App.Runtime layout because we want both and App.Ref project depends on the
+        App.Runtime project.
+      -->
+      <_AppRefPathExists Condition=" EXISTS('$(_AppRefPath)') ">true</_AppRefPathExists>
+      <_AppRefPathExists Condition=" '$(_AppRefPathExists)' == '' ">false</_AppRefPathExists>
     </PropertyGroup>
 
-    <!-- Use package because .dotnet/ folder doesn't contain RuntimeList.xml file and SharedFxTests checks that. -->
-    <Unzip
-       SourceFiles="$(RepoRoot)artifacts\packages\$(Configuration)\Shipping\Microsoft.AspNetCore.App.Runtime.$(TargetRuntimeIdentifier).$(SharedFxVersion).nupkg"
-       DestinationFolder="$(OutputPath)\SharedFx.Unzip" />
-    <ItemGroup>
-      <_appRuntimeFiles Include="$(OutputPath)\SharedFx.Unzip\**\*.txt" />
-      <_appRuntimeFiles Include="$(OutputPath)\SharedFx.Unzip\**\*.json" />
-      <_appRuntimeFiles Include="$(OutputPath)\SharedFx.Unzip\**\*.dll" />
-      <_appRuntimeFiles Include="$(OutputPath)\SharedFx.Unzip\**\RuntimeList.xml" />
-    </ItemGroup>
-    <Copy SourceFiles="@(_appRuntimeFiles)"
-        DestinationFolder="$(OutputPath)\SharedFx.Layout\shared\Microsoft.AspNetCore.App\$(SharedFxVersion)" />
-    <Copy SourceFiles="$(OutputPath)\SharedFx.Unzip\Microsoft.AspNetCore.App.versions.txt"
-        DestinationFiles="$(OutputPath)\SharedFx.Layout\shared\Microsoft.AspNetCore.App\$(SharedFxVersion)\.version" />
-
-    <Unzip Condition="Exists('$(RepoRoot)artifacts\packages\$(Configuration)\Shipping\Microsoft.AspNetCore.App.Ref.$(SharedFxVersion).nupkg')"
-      SourceFiles="$(RepoRoot)artifacts\packages\$(Configuration)\Shipping\Microsoft.AspNetCore.App.Ref.$(SharedFxVersion).nupkg"
-      DestinationFolder="$(OutputPath)\SharedFx.Layout\packs\Microsoft.AspNetCore.App.Ref\$(SharedFxVersion)" />
-    <ZipDirectory Condition="Exists('$(OutputPath)\SharedFx.Layout')"
-      SourceDirectory="$(OutputPath)\SharedFx.Layout"
-      DestinationFile="$(OutputPath)\SharedFx.Layout.zip" Overwrite="true" />
-
-    <ItemGroup>
-      <HelixCorrelationPayload Include="$(OutputPath)\SharedFx.Layout.zip" Destination="$(DotNetCliDestination)" />
+    <!-- After complaining about missing App.Ref layout, do nothing else in this target.  -->
+    <Warning Text="Targeting path location '$(_AppRefPath)' does not exist."
```

Comment:
@BrennanConroy and others: Any thoughts on whether this should instead be a visible `<Message Importance="High" ... />` instead❔

Reply:
Should probably change the message to mention that the test run wont be using the local build of our shared fx and will use XYZ instead

Reply:
This isn't a .NET SDK project and would need to read artifacts/bin/GenerateFiles/* to figure out the default when 7.0.0-ci isn't available. How about

```suggestion
    <Warning Text="Targeting path location '$(_AppRefPath)' does not exist and tests will use the SDK's bundled runtimes and targeting pack."
```

or (same wording but answering my own question to avoid the warnings-as-error elevation)

```suggestion
    <Message Importance="High" Text="Targeting path location '$(_AppRefPath)' does not exist and tests will use the SDK's bundled runtimes and targeting pack."
```

Could also add that tests which absolutely require the just-built runtime (most obviously the project template tests) are doomed to fail.

Reply:
Thanks for the thumbs up @BrennanConroy. Do you prefer `<Warning ... />` or `<Message Importance="High" ... />`❔

Reply:
If this happens on the CI do we want to know about it? Then warning. Otherwise, whichever is fine.

Reply:
> If this happens on the CI do we want to know about it?

Yes because it dooms a subset of the tests to fail and it's not worth running the rest. I'm going to use slightly stronger language in the `<Error/>` two lines down.

To clarify: The `<Error/>` happens on the CI or if you add `/p:DoNotRequireSharedFxHelix=false` to a RunHelix.ps1 command.